### PR TITLE
Fix a bug in encodeFinalFrame

### PR DESCRIPTION
**Pull request by princemaple**
Hi @chris-rudmin,

Thanks for the awsome work. Loving it.

Please take a look at this PR, thanks.
It fixes a type mismatch, getting a Float32Array while expecting an Array of Float32Array.

Not sure why it ever worked, if it did.
Tested on latest Firefox and Chrome.

### CODE REVIEW

**Comment by chris-rudmin** [comment]
Cool. Thanks for the fix.  We were probably not getting the data in the last frame which was less than 20ms.
I think the array length should actually be the number of channels. I'll take a look when i'm at home.

**Comment by princemaple** [comment]
You are right. It should be array of multiple float32arrays when it has
multiple channels.

It threw an exception for me without the fix. That's strange that it worked
for you.

Thanks!

BTW I'm in Australia, bed time now, I can fix my PR tomorrow unless you
want to take ovet.

**Comment by chris-rudmin** [comment]
I'm away from home without my computer, so take your time.

**Comment by princemaple** [comment]
Hi @chris-rudmin 

I've updated the PR so now it works with multiple channels.
I used ES6 `Array.prototype.fill` in the code since the browsers that support audio recording all seem to support `.fill`

I wasn't sure how your coding style (space inside parenthesis) would work with multi-line or very long line of code, so ended up using my own coding style. If you do mind, please suggest how you would write the code, I'll update the code accordingly.

Other than that, it should be all good now.

Po

File: oggopusEncoder.js [hunk 1]
```diff
@@ -78,7 +78,9 @@ OggOpusEncoder.prototype.encode = function( buffers ) {
 };
 
 OggOpusEncoder.prototype.encodeFinalFrame = function() {
-  this.encode( new Float32Array( this.encoderBufferLength - this.encoderBufferIndex ) );
+  this.encode(new Array(this.numberOfChannels).fill().map(function() {
+    return Float32Array(this.encoderBufferLength - this.encoderBufferIndex);
```
Comment (chris-rudmin) [inline oggopusEncoder.js]: Do we need the `new` keyword here?

Comment (princemaple) [inline oggopusEncoder.js]: Yes we do, I can't believe how sloppy I've been on this PR.
It works without for now but will be completed broken in ES6.

**Comment by princemaple** [comment]
Sorry about that, updated.

File: oggopusEncoder.js (hunk 1 above)
Comment (chris-rudmin) [inline oggopusEncoder.js]: This throws an error for me `(intermediate value).fill is not a function`.  Using Chrome ver 43

Comment (princemaple) [inline oggopusEncoder.js]: https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/fill
Failed to spot the part says it requires a flag to work on Chrome. I'll go with the old way.

**Comment by princemaple** [comment]
Updated and rebased. Sorry about all the fuss.

**Comment by chris-rudmin** [comment]
No worries, thanks for taking the time to do a good PR. :) Since we went through all that trouble, i'm gonna be picky. I actually prefer the solution with `.map`. I think it was a bit cleaner.

` new Array(this.numberOfChannels).map(function(){ return new Float32Array(...) });`

**Comment by chris-rudmin** [comment]
Actually, i tried this and it does not work cleanly. Sorry to flip flop.  I'm gonna take your pull as it is now.

**Comment by princemaple** [comment]
`new Array(NUMBER).map` is something we all wanted. :)
I think ES6 `new Array(NUMBER).fill().map` is the closest we can get, if only Chrome enabled it by default...

**Pull request merged**